### PR TITLE
Shrike's preparation logic generates environment.yml file for Conda-native dependencies

### PR DESCRIPTION
**Pull request by haozhestat**
To put Conda deps in environment.yml and put pip deps in requirements.txt.

### CODE REVIEW

File: shrike/build/commands/prepare.py [hunk 1]
```diff
@@ -619,63 +619,72 @@ def is_in_additional_includes(
...
-                    file.write(req)
-                    if not req.endswith("\n"):
-                        file.write("\n")
+            if pip_dependencies:
+                log.info(
+                    f"Found pip dependencies for component {component_name} in {component_repo}. Writing to requirements.txt."
+                )
+                with open(os.path.join(cur_path, "requirements.txt"), "w") as file:
+                    for req in pip_dependencies:
+                        file.write(req)
```
Comment (dkmiller) [inline shrike/build/commands/prepare.py:660]: Suggestion: use the `writelines` method.

**Review by dkmiller — CHANGES_REQUESTED** [review]
One minor comment, also a bit concerned by the drop in code coverage. Can you add some additional tests? If it's hard to think of tests to add, probably an indication you should split up the code...

File: tests/tests_build/commands/test_prepare.py [hunk 2]
```diff
@@ -967,18 +967,22 @@ def test_create_requirements_file_with_duplicate_component_name(caplog):
...
     component_files = [tmp_dir + "/spec.yaml", tmp_dir + "/spec_dup.yaml"]
     with caplog.at_level("INFO"):
-        id = prep._create_requirements_files(component_files)
+        id = prep._create_dependencies_files(component_files)
     component_dependencies_repo = "component_dependencies_" + id
     assert os.path.exists(component_dependencies_repo + "/dummy/requirements.txt")
     assert os.path.exists(
         component_dependencies_repo + "/dummy_spec_dup/requirements.txt"
     )
+    assert os.path.exists(component_dependencies_repo + "/dummy/environment.yml")
```
Comment (fuhuifang) [inline tests/tests_build/commands/test_prepare.py:976]: Not sure if this helps with coverage rate, but you could try adding "assert not exist" for environment.yml before calling `_create_dependencies_files()`.

**Review by fuhuifang — APPROVED** [review]

**Review by dkmiller — APPROVED** [review]

**Comment by haozhestat** [comment]
> One minor comment, also a bit concerned by the drop in code coverage. Can you add some additional tests? If it's hard to think of tests to add, probably an indication you should split up the code...

@dkmiller: I just checked the coverage report ([Codecov](https://app.codecov.io/gh/Azure/shrike/compare/256/overview)). After merging this PR into main, the coverage rate will increase 0.08%.

**Pull request merged**